pkg/gardenlet/controller/seed/seed_care_control.go

```diff
@@ -0,0 +1,208 @@
+		// correct types will be updated, and any other conditions will remain intact
+		conditions := buildSeedConditions(seed.Status.Conditions, updatedConditions, conditionTypes)
+		log.Info("Updating seed status conditions and constraints")
+		if err := patchSeedStatus(ctx, gardenClient, seed, conditions); err != nil {
+			log.Error(err, "Could not update Seed status")
+			return nil // We do not want to run in the exponential backoff for the condition checks.
+		}
+	}
+	return nil
+}
+
+func careSetupFailure(ctx context.Context, gardenClient client.Client, seed *gardencorev1beta1.Seed, message string, conditions []gardencorev1beta1.Condition) error {
+	updatedConditions := make([]gardencorev1beta1.Condition, 0, len(conditions))
+	for _, cond := range conditions {
+		updatedConditions = append(updatedConditions, gardencorev1beta1helper.UpdatedConditionUnknownErrorMessage(cond, message))
+	}
+
+	if !gardencorev1beta1helper.ConditionsNeedUpdate(conditions, updatedConditions) {
+		return nil
+	}
+
+	return patchSeedStatus(ctx, gardenClient, seed, updatedConditions)
+}
+
+// buildSeedConditions builds and returns the seed conditions using the given seed conditions as a base,
+// by first removing all conditions with the given types and then merging the given conditions (which must be of the same types).
+func buildSeedConditions(seedConditions []gardencorev1beta1.Condition, conditions []gardencorev1beta1.Condition, conditionTypes []gardencorev1beta1.ConditionType) []gardencorev1beta1.Condition {
+	result := gardencorev1beta1helper.RemoveConditions(seedConditions, conditionTypes...)
+	result = gardencorev1beta1helper.MergeConditions(result, conditions...)
+	return result
+}
+
+func patchSeedStatus(ctx context.Context, c client.StatusClient, seed *gardencorev1beta1.Seed, conditions []gardencorev1beta1.Condition) error {
+	patch := client.StrategicMergeFrom(seed.DeepCopy())
+	seed.Status.Conditions = conditions
+	return c.Status().Patch(ctx, seed, patch)
+}
+
+func (r *careReconciler) conditionThresholdsToProgressingMapping() map[gardencorev1beta1.ConditionType]time.Duration {
+	out := make(map[gardencorev1beta1.ConditionType]time.Duration)
+	for _, threshold := range r.config.Controllers.SeedCare.ConditionThresholds {
+		out[gardencorev1beta1.ConditionType(threshold.Type)] = threshold.Duration.Duration
+	}
+	return out
+}
+
+func (c *Controller) seedCareAdd(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+	c.seedCareQueue.Add(key)
+}
+
+func (c *Controller) seedCareUpdate(_, newObj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(newObj)
+	if err != nil {
+		return
+	}
+
+	oldSeed, ok := newObj.(*gardencorev1beta1.Seed)
+	if !ok {
+		return
+	}
+
+	newSeed, ok := newObj.(*gardencorev1beta1.Seed)
+	if !ok {
+		return
+	}
+
+	if newSeed.Generation == newSeed.Status.ObservedGeneration && apiequality.Semantic.DeepEqual(oldSeed.Status, newSeed.Status) {
```

Comment:
`ObservedGeneration` is "owned" by the main reconciler and is not of concern to this secondary reconciler.
We should do something similar like in the shoot care reconciler, where shoots are enqueued on updates if the reconciling just finished successfully.
I.e., let's enqueue seeds if their `Bootstrapped` condition status changed.

Reply:
I have added a check if for the conditions, I hope that I understood you properly